### PR TITLE
VR: Change Get VRC to an func and add test

### PR DESCRIPTION
**Pull request by iamniting**
Implement a func to get the volume replica class and write the unit test to test it.

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: controllers/vrc.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
```
Comment (Madhu-1) [inline controllers/vrc.go:1]: rename the file to volumereplicationclass.go

File: controllers/vrc.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import (
+	"context"
+
+	replicationv1alpha1 "github.com/kube-storage/volume-replication-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r VolumeReplicationReconciler) getVolumeReplicaCLass(vrcName string) (*replicationv1alpha1.VolumeReplicationClass, error) {
+
```
Comment (Madhu-1) [inline controllers/vrc.go:28]: remove this new line?

**Comment by Madhu-1** [comment]
@iamniting update the PR description and commit message currently it's empty

**Comment by iamniting** [comment]
> @iamniting update the PR description and commit message currently it's empty

Done

File: controllers/volumereplicationclass.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+func (r VolumeReplicationReconciler) getVolumeReplicaCLass(vrcName string) (*replicationv1alpha1.VolumeReplicationClass, error) {
+	vrcObj := &replicationv1alpha1.VolumeReplicationClass{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: vrcName}, vrcObj)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Error(err, "VolumeReplicationClass not found", "VolumeReplicationClass", vrcName)
+		}
+		return nil, err
```
Comment (Madhu-1) [inline controllers/volumereplicationclass.go:34]: don't we need to log if the error is other than not found?

**Review by sp98 — APPROVED** [review]

File: controllers/volumereplicationclass.go (hunk 3 above)
Comment (iamniting) [reply controllers/volumereplicationclass.go:36]: We should

**Pull request merged**